File: notification-service/src/main/java/com/ecommerce/notification_service/model/Notification.java
package com.ecommerce.notification_service.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private Long orderId;

    @Column(nullable = false)
    private Long paymentId;

    @Column(nullable = false)
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationType type;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Builder.Default
    @Column(nullable = false)
    private boolean read = false;

    @PrePersist
    public void prePersist() {
        if (type == null) {
            type = NotificationType.PAYMENT_SUCCESS;
        }
        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/app/core/models/notification.ts
export type NotificationType =
  | 'PAYMENT_SUCCESS'
  | 'ORDER_CONFIRMED'
  | 'PAYMENT_REFUNDED'
  | 'ORDER_CANCELLED'
  | 'ORDER_SHIPPED'
  | 'ORDER_DELIVERED';

export interface NotificationResponse {
  id: number;
  userId: number;
  orderId: number | null;
  paymentId: number | null;
  message: string;
  type: NotificationType;
  createdAt: string;
  read: boolean;
}
